be/src/vec/exec/format/parquet/vparquet_reader.cpp

```diff
@@ -545,14 +545,18 @@ Status ParquetReader::_process_page_index(const tparquet::RowGroup& row_group) {
     if (!_has_page_index(row_group.columns, page_index)) {
         return Status::OK();
     }
-    int64_t buffer_size = page_index._column_index_size + page_index._offset_index_size;
-    uint8_t buff[buffer_size];
+    //    int64_t buffer_size = page_index._column_index_size;
+    uint8_t col_index_buff[page_index._column_index_size];
     int64_t bytes_read = 0;
-    RETURN_IF_ERROR(
-            _file_reader->readat(page_index._column_index_start, buffer_size, &bytes_read, buff));
-
+    RETURN_IF_ERROR(_file_reader->readat(page_index._column_index_start,
```

Comment:
I think we can merge this 2 `readat()`, and split the buffer into 2 buffers, so that we can save one remote file read call.